workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerBase.cs

```diff
@@ -49,11 +49,25 @@ public void Dispose()
             World.Dispose();
         }
 
+        public bool RequiresUniqueWorkerIdPerConnection = false;
+
         public virtual void Connect(ConnectionConfig config)
         {
             if (config is ReceptionistConfig)
             {
-                Connection = ConnectionUtility.ConnectToSpatial((ReceptionistConfig) config, GetWorkerType, WorkerId);
+                string workerIdForConnection;
+
+                if (RequiresUniqueWorkerIdPerConnection)
+                {
+                    workerIdForConnection = $"{WorkerId}-{Guid.NewGuid()}";
```

Comment:
If you do this then `connection.GetWorkerId()` and `WorkerId` will have different values because you never set the `WorkerId`, I think this is very unexpected behaviour.

Reply:
`WorkerId` is currently only used privately within `WorkerBase`, so I'm thinking of making the whole property a private string instead. 

And we you want to get "the worker id" we just get it from the connection.

Reply:
Why wouldn't we rather use: $"{GetWorkerType}-{Guid.NewGuid()}" and then set that to WorkerId? This way we can keep the WorkerId field and the change shouldn't be major.

Reply:
Yes that is a better idea :D